examples/query/src/main.rs

```diff
@@ -1,3 +1,63 @@
+use std::process::exit;
+
+use libbpf_rs::query;
+use nix::unistd::Uid;
+use structopt::StructOpt;
+
+/// Query the system about BPF-related information
+#[derive(Debug, StructOpt)]
+enum Command {
+    /// Display information about progs
+    Prog,
+    /// Display information about maps
+    Map,
+    /// Display information about BTF
+    Btf,
+}
+
+fn prog() -> i32 {
+    for prog in query::ProgInfoIter::default() {
+        println!(
+            "name={:<16} type={:<15} run_count={:<2} runtime_ns={}",
+            prog.name,
+            prog.ty.to_string(),
+            prog.run_cnt,
+            prog.run_time_ns
+        );
+    }
+
+    0
```

Comment:
why all these zeroes? look weird and it's the only value you can return here (btf, how can iterators communicate failure?) If it's an example, I'd keep it to bare minimum  and just return 0 in main().

Reply:
Could communicate errors by having iterator return `Result` type. But I think for the libbpf query interface it should be ok to return bare values. I don't see any errors that don't terminate iteration.

Reply:
my point was that when one of those kernel APIs does return real error, how does iterator communicate that iteration process stopped due to error, not because we finished iterating all of the items? It's a general question about Rust iteration, how to distinguish premature iteration end due to error vs completed iteration?

Reply:
I'm not sure there's a great way. I guess one way to do that would have a constructor that returns a `Result<*InfoIter>`. But then you'd have to iterate everything ahead of time and store the data instead of get it on demand.